autobahn/src/main/java/io/crossbar/autobahn/wamp/reflectionRoles/ArgumentUnpacker.java

```diff
@@ -0,0 +1,51 @@
+package io.crossbar.autobahn.wamp.reflectionRoles;
+
+import io.crossbar.autobahn.wamp.interfaces.ISerializer;
+
+import java.lang.reflect.Parameter;
+import java.util.List;
+import java.util.Map;
+
+public class ArgumentUnpacker {
+    private final ParameterInfo[] mParameters;
+
+    public ArgumentUnpacker(Parameter[] parameters) {
+        mParameters = new ParameterInfo[parameters.length];
+
+        for (int i = 0; i < parameters.length; i++) {
+            Parameter parameter = parameters[i];
+            String parameterName = parameter.getName();
```

Comment:
For Android, the main issue is this API(level 26 only) and of course its use is central to what we are trying to achieve here.